typify/tests/schemas/property-pattern.json

```diff
@@ -0,0 +1,342 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "tree-sitter grammar specification",
+  "type": "object",
+  "required": [
+    "name",
+    "rules"
+  ],
+  "additionalProperties": false,
+  "properties": {
+    "name": {
+      "description": "the name of the grammar",
+      "type": "string",
+      "pattern": "^[a-zA-Z_]\\w*"
+    },
+    "inherits": {
+      "description": "the name of the parent grammar",
+      "type": "string",
+      "pattern": "^[a-zA-Z_]\\w*"
+    },
+    "rules": {
+      "type": "object",
+      "patternProperties": {
+        "^[a-zA-Z_]\\w*$": {
+          "$ref": "#/definitions/rule"
+        }
+      },
+      "additionalProperties": false
+    },
```

Comment:
is this the only significant part of the test? Can we remove the other properties?

Reply:
It is, I've reduced this test case.